packages/devextreme/js/__internal/core/state_manager/setup_state_manager.ts

```diff
@@ -0,0 +1,81 @@
+/* eslint-disable spellcheck/spell-checker */
+import type { DIContext } from '../di';
+import { Logger } from './logger';
+import { StateManagerFactory } from './state_manager';
+import type * as StateManagementTypes from './types';
+
+export type DecoratorFunction<T = unknown> = (instance: T) => T;
+
+export interface StateManagerInitializerOptions {
+  logLevel?: StateManagementTypes.LogLevel;
+  componentName?: string;
+  diContext?: DIContext;
```

Comment:
Without `componentName` and `diContext` options the `setupStateManager` function throw runtime error
Let's make this fields required in this interface:
```suggestion
  componentName: string;
  diContext: DIContext;
```

Reply:
Changed